x86_pvh_dom0.md

```diff
@@ -0,0 +1,176 @@
+```
+and change default "debian" to "dom0"
+
+## Altering the Domain-0 root file system 
+In case you need to modify the root file system use the following commands to mount and umount:
+```sh
+sudo guestmount -a dom0.qcow2 -i mnt/
+sudo guestunmount mnt
+```
+
+## Update default Debian environment
+
+Update the kernel to more recent one:
+```sh
+apt update
+apt-get install linux-image-5.10.0-0.bpo.8-amd64 linux-headers-5.10.0-0.bpo.8-amd64 
+```
+
+Create a new user:
+```sh
+adduser a2k
+usermod -aG sudo a2k
+getent group sudo
+```
+
+Allow password authentication for sshd:
+```sh
+vi /etc/ssh/sshd_config
+PasswordAuthentication yes
+dpkg-reconfigure openssh-server
+systemctl restart ssh
+```
+
+And connect to the QEMU with ssh:
+```sh
+ssh a2k@localhost -p 2222
+```
+
+## Install build essentials
+To be able to build the essential software install the following packages:
+```sh
+sudo apt-get install git
+sudo apt-get install build-essential linux-source bc kmod cpio flex libncurses5-dev libelf-dev libssl-dev
+sudo apt-get build-dep xen python3-dev ninja-build libsystemd-dev 
+```
+
+## Build and install Xen
+```sh
+a2k@debian:~/projects/
+git clone https://xenbits.xen.org/git-http/xen.git
+git checkout origin/stable-4.15 --track
+./configure --disable-docs --disable-stubdom --prefix=/usr/local --libdir=/usr/lib --enable-systemd
+make -j4
+make debball
+```
+
+And install it
+```sh
+sudo dpkg -i dist/xen-upstream-4.15.1-pre.deb
+
+sudo systemctl enable xen-qemu-dom0-disk-backend.service
+sudo systemctl enable xen-init-dom0.service
+sudo systemctl enable xenconsoled.service
+sudo systemctl enable xendomains.service
+sudo systemctl enable xen-watchdog.service
+sudo systemctl enable xendriverdomain.service
+```
+
+Edit /etc/default/grub and update the command line for Xen and the kernel:
+```
+GRUB_CMDLINE_LINUX="console=tty0 console=ttyS0,115200 earlyprintk=ttyS0,115200 console=hvc0"
+GRUB_CMDLINE_XEN="iommu=no-sharept,verbose,debug dom0_mem=4096M,max:4096M debug=y loglvl=all guest_loglvl=all com1=115200,8n1 console=com1,vga sync_console_to_ring=true sync_console"
+```
+For PVH Dom0 add "dom0=pvh" to the `GRUB_CMDLINE_XEN` command line.
+Set `GRUB_DEFAULT=2` for Xen to boot first
+
+Update the grub and restart:
+```sh
+sudo update-grub
+sudo reboot
+```
+Test that everything is set up right after the reboot:
+```sh
+sudo xl info
+```
+During Xen development use the following to test the changes:
+```sh
+CC=gcc make -C xen/ -j4
+sudo cp xen/xen.gz /boot/xen-4.15.gz
+sudo reboot
+```
+## Download DomU's root file system
+```sh
+wget https://cloud.debian.org/images/cloud/buster/20210721-710/debian-10-nocloud-amd64-20210721-710.qcow2
+ln -s debian-10-nocloud-amd64-20210721-710.qcow2 domu.qcow2
+```
+## Donwload and build DomU's Linux kernel
+```sh
+git clone https://github.com/torvalds/linux.git
+git checkout v5.10
+make x86_64_defconfig
+make xen.config
+```
+Update the configuration to built in the network and block device drivers:
```

Comment:
"to built"? It could be good to reword this part.

Reply:
Ok